coalib/results/SourceRange.py

```diff
@@ -120,6 +120,47 @@ def expand(self, file_contents):
                                        tr.end.line,
                                        tr.end.column)
 
+    @enforce_signature
+    def affected_source(self, file_dict: (dict, None)):
+        r"""
+        Tells which are the lines affected in the file within a specified
+        range.
+
+        >>> from os.path import abspath
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=2)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n')})
+        ['    x = 2  \n']
+
+        If there is no file to search for the contents then it will
+        produce None as the result.
+
+        >>> sr.affected_source(None)
+        >>>
+
+        If more than one line is infected.
+
+        >>> sr = SourceRange.from_values('file_name', start_line=2, end_line=3)
+        >>> sr.affected_source({abspath('file_name'):
+        ...     ('def fun():\n', '    x = 2  \n', '    print(x)  \n')})
+        ['    x = 2  \n', '    print(x)  \n']
+
+        :param file_contents:
+            The content of the file. Each entry is one of the line of the
+            file in sequence.
+        :return:
+            List of lines that are affected in the specified file.
+            If the file is not affected return None
+        """
+        if file_dict:
+            source_lines = []
+            file_contents = file_dict[self.start.file]
+            if self.start.line and self.end.line:
+                # line number starts from 1, index starts from 0
+                for line_number in range(self.start.line - 1, self.end.line):
+                    source_lines.append(file_contents[line_number])
```

Comment:
`yield file_contents[line_number]` generates one result upon one call. For `range.start.line=3` and `range.end.line=5` we have to call the generator `3` times to get  the affected source lines. That means again a for loop in `ConsoleInteraction.py`. Is it ok ? I was printing the `list` of the affected lines. May be you are telling to print the *each* source line on its *own* line. Should we print each source line on its own line ? Is it something else you are saying ?